plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java

```diff
@@ -963,4 +965,20 @@ private static List<LocatedFileStatus> remoteIteratorToList(RemoteIterator<Locat
         }
         return result;
     }
+
+    @Test
+    public void testAnonymousAccess()
+            throws Exception
+    {
+        Configuration config = newEmptyConfiguration();
+
+        config.set(S3_ANONYMOUS_REQUESTS_ENABLED, "true");
+
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            fs.initialize(new URI("s3n://test-bucket/"), config);
+            AWSCredentialsProvider provider = getAwsCredentialsProvider(fs);
+            assertInstanceOf(provider, AWSStaticCredentialsProvider.class);
+            assertInstanceOf(provider.getCredentials(), AnonymousAWSCredentials.class);
```

Comment:
Just verifying an instance isn't helpful for the actual environment. We should make sure if we can access public S3 bucket.